bothub-nlp-nlu-worker/bothub_nlp_nlu_worker/bothub_nlp_nlu/debug_parse.py

```diff
@@ -0,0 +1,106 @@
+
+            idx_dict = {}  # fixing intent name to a index ex: {'violence': 0, 'immigration': 1, ... }
+            size = len(self.intention_names)
+            for i in range(size):
+                idx_dict[self.intention_names[i]] = i
+
+            intent_list = [0] * len(self.intention_names)
+            intent_name_list = [''] * len(self.intention_names)
+            size = len(result_json['intent_ranking'])
+            for i in range(size):
+                intent_name = result_json['intent_ranking'][i]['name']
+                intent_list[idx_dict[intent_name]] = result_json['intent_ranking'][i]['confidence']
+                intent_name_list[idx_dict[intent_name]] = result_json['intent_ranking'][i]['name']
+
+            prob_array = np.array(intent_list)
+            prob_array = prob_array.reshape((1, len(intent_list)))
+            result_list.append(prob_array)
+
+        result_array = result_list[0]
+        for i in range(1, len(result_list)):
+            result_array = np.vstack([result_array, result_list[i]])
+        return result_array
+
+    def get_result_per_word(self, text, num_samples):
+        explainer = LimeTextExplainer(class_names=self.intention_names)
+        labels = list(range(len(self.intention_names)))  # List
+        exp = explainer.explain_instance(text, self.parse, num_features=6, labels=labels, num_samples=num_samples)
+        result_per_word = {}
+        for i in labels:
+            for j in exp.as_list(label=i):
+                if j[0] not in result_per_word:
+                    result_per_word[j[0]] = []
+                result_per_word[j[0]].append({'intent': self.intention_names[i], 'relevance': j[1] * 100})
+        for word in result_per_word:
+            result_per_word[word] = sorted(result_per_word[word], key=lambda k: k['relevance'], reverse=True)
+        print(json.dumps(result_per_word, indent=2))
+        return result_per_word
+
+    def get_result_per_intent(self, text, num_samples):
+        explainer = LimeTextExplainer(class_names=self.intention_names)
+        labels = list(range(len(self.intention_names)))  # List
+        exp = explainer.explain_instance(text, self.parse, num_features=6, labels=labels, num_samples=num_samples)
+        result_per_intent = {}
+        for intent in self.intention_names:
+            result_per_intent[intent] = []
+        for i in labels:
+            intent_sum = 0
+            for j in exp.as_list(label=i):
+                result_per_intent[self.intention_names[i]].append({'word': j[0], 'relevance': j[1] * 100})
+                intent_sum += j[1]
+            result_per_intent[self.intention_names[i]].append({'sum': intent_sum, 'relevance': -1})
+        for intent in result_per_intent:
+            result_per_intent[intent] = sorted(result_per_intent[intent], key=lambda k: k['relevance'], reverse=True)
+
+        return result_per_intent
+
+
+def get_intention_list(repository_authorization):
```

Comment:
If the backend return has a key detail, will you not return anything in the method? possibly going wrong, perform the tests with invalid intelligence.